src/os/FileJournal.cc

```diff
@@ -1242,7 +1249,7 @@ void FileJournal::write_thread_entry()
     }
     
 #ifdef HAVE_LIBAIO
-    if (aio) {
+    if (aio && !g_conf->journal_dynamic_throttle) {
```

Comment:
@athanatos Here is the reason.

1. I have disabled all other throttle scheme if user enabled dynamic throttle. It is not holding any throttling related locks any more. So, throttle_bytes is not available here.

2. In my test I didn't see any advantage by enabling/disabling this portion of the code. In fact, it is hardly going into aio_cond.Wait part. Even if it is going in some other test scenario, I feel this will induce a spikiness in the write behavior.

3. Since journal write path is executing without any throttle here (and thus very fast), we need to limit the coalescing (with reduced journal_max_write_entries) so that it doesn't go far ahead from backend Tx execution. In that case, I felt the above code block would be redundant.

4. Yes, if journal is empty, dyn throttle will not be doing anything. But, don't you think journal is almost empty and lot pore pending aios are contradictory ? If there are lot of pending aios, the in memory calculation of circular buffer should grow (showing journal usage) as it is not giving enough job to the filestore queue for Tx execution.

5. I felt, even if in some scenarios it generates lot of aio , controlling it from application and generating spiky behavior probably unnecessary. There are kernel block device params user can set in Linux if they feel the backend storage is not capable of handling bigger aio QD.

Let me know if I am missing anything.